lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -458,70 +485,66 @@ class TouchAdapter extends BaseAdapter {
     ui.PointerChange change,
     html.TouchEvent event,
   ) {
-    final html.TouchList touches = event.changedTouches;
     final List<ui.PointerData> data = List<ui.PointerData>();
-    final int len = touches.length;
-    for (int i = 0; i < len; i++) {
-      final html.Touch touch = touches[i];
+    for (html.Touch touch in event.changedTouches) {
```

Comment:
The reason we avoided `for in` here was to avoid the creation of unnecessary iterators.

Reply:
Are we worrying about performance? I think it's a premature optimization unless it's backed by a benchmark. In my opinion, loops in this file is of the same impact as loops in app files, so if app files are using `for in` everywhere, so should we.

Reply:
I agree it's premature optimization and it's not backed by a benchmark. This specific code path though, could be a gazillion times when the user is moving their fingers. And we know for sure that Dart generates sub-optimal JS code in some cases for `for in`. That's why we avoided it here.

You can keep the `for in` if you want. We are gonna have to tackle performance at some point anyway. Then, we would more concrete data to optimize against.

Reply:
Ok, I'll leave it to you. Thanks for explaining.